torchio/transforms/transform.py

```diff
@@ -62,7 +63,22 @@ def __call__(self, data: TypeTransformInput) -> TypeTransformInput:
                 a tensor, the affine matrix is an identity and a tensor will be
                 also returned.
         """
+        #Execution's seed
+        if not seed:
+            seed = gen_seed()
+
+        #Store the current rng_state to reset it after the execution
+        torch_rng_state = torch.random.get_rng_state()
+        if isinstance(seed, int):
+            torch.manual_seed(seed=seed)
+
+        self.transform_params = {}
+        self._store_params()
+        self.transform_params["seed"] = seed
```

Comment:
Can the seed stuff be moved into `RandomTransform` instead?

Reply:
not sure, there are 2 functionality here : handeling the seed, which is only related to randomTransform but also store transform param, with _store_params which then need to be in transform.py

I let ghiles explain how the seeding works, but for the parameter storing it simply add all parameters of the class. This make it convenient to add any require parameter, 
for instance in the random_affine.py I just had to add:
        self.apply_scales =  scaling_params
        self.apply_rot = rotation_params
        self.apply_trans = translation_params
(and now we know it is not enough I could add
self.afffine = ...

I know we did not agreed on previous discussion on which parameters needs to be include in the history, but at least with this mechanism, it make it easy to modify torchio transform to add any need information ... non ?

Reply:
Hmm ok, I think I see why it's not possible to move to `RandomTransform`. I hope we'll be able to come up with cleaner solutions for this soon.

Reply:
may be the seeding part to random transform and _store_params() to transform.py .. @GReguig ?

Reply:
It is clearly possible to do as @romainVala is suggesting, treating the seeding part in the 'RandomTransform' part and executing the '_store_params' in the 'Transform'. However, it introduces an inconsistency in the different types of transforms and will force very specific treatments when wanting to loop over a sequence of transforms. This would be problematic if we want to reproduce a whole Compose, some transforms will have a seed and others won't. 

I personally think it would be best to keep the consistency (with a seed for every transform, even if it is not a random one) rather than handling very specific treatments afterwards. What do you think ?

Reply:
I just think it doesn't make a lot of sense to store the seed for a transform that is not random. I'm also not sure that the effort to reproduce a composed transform is necessary. You could just loop over the history and applied one transform after the other, right?

Reply:
I agree, 
it may then worth to add the specific treatments in the code, in order to get a lighter (and then cleaner history)